crates/fluvio-spu/src/services/public/mod.rs

```diff
@@ -145,3 +175,33 @@ impl FluvioService for PublicService {
         Ok(())
     }
 }
+
+async fn send_private_request_to_leader<R: Request>(
+    ctx: &DefaultSharedGlobalContext,
+    replica_id: &ReplicaKey,
+    req: R,
+) -> Result<R::Response, ErrorCode> {
+    let spu = match ctx.replica_localstore().spec(replica_id) {
+        Some(replica) => replica.leader,
+        None => return Err(ErrorCode::TopicNotFound),
+    };
+    let Some(spu_spec) = ctx.spu_localstore().spec(&spu) else {
+        return Err(ErrorCode::SpuNotFound);
+    };
+    let leader_endpoint = spu_spec.private_endpoint.to_string();
+    debug!(
+        spu,
+        leader_endpoint, "send private request to replica leader"
+    );
+    let mut socket = FluvioSocket::connect(&leader_endpoint)
```

Comment:
this should be using private API to SPU, otherwise this will incur connection latency

Reply:
This is using private api:
```
let leader_endpoint = spu_spec.private_endpoint.to_string();
```